jetty-http2/http2-http-client-transport/src/test/java/org/eclipse/jetty/http2/client/http/MultiplexedConnectionPoolTest.java

```diff
@@ -0,0 +1,304 @@
+            client = null;
+        }
+    }
+
+    @Test
+    public void testMaxDurationConnectionsWithMultiplexedPool() throws Exception
+    {
+        // ConnectionPools may open a few more connections than expected.
+
+        final int maxDuration = 30;
+        AtomicInteger poolAcquireCounter = new AtomicInteger();
+        AtomicInteger poolRemoveCounter = new AtomicInteger();
+        AtomicReference<Pool<Connection>> poolRef = new AtomicReference<>();
+        ConnectionPoolFactory factory = new ConnectionPoolFactory("duplex-maxDuration", destination ->
+        {
+            int maxConnections = destination.getHttpClient().getMaxConnectionsPerDestination();
+            Pool<Connection> pool = new Pool<>(Pool.StrategyType.FIRST, maxConnections, false);
+            poolRef.set(pool);
+            MultiplexConnectionPool connectionPool = new MultiplexConnectionPool(destination, pool, destination, 2)
+            {
+                @Override
+                protected void acquired(Connection connection)
+                {
+                    poolAcquireCounter.incrementAndGet();
+                }
+
+                @Override
+                protected void removed(Connection connection)
+                {
+                    poolRemoveCounter.incrementAndGet();
+                }
+            };
+            connectionPool.setMaxDuration(maxDuration);
+            return connectionPool;
+        });
+
+        startServer(new EmptyServerHandler());
+
+        QueuedThreadPool clientThreads = new QueuedThreadPool(3);
+        clientThreads.setName("client");
+        HttpClientTransport transport = new HttpClientTransportOverHTTP2(new HTTP2Client());
+        transport.setConnectionPoolFactory(factory.factory);
+        client = new HttpClient(transport, null);
+        client.setExecutor(clientThreads);
+        client.start();
+
+        // Use the connection pool 5 times with a delay that is longer than the max duration in between each time.
+        for (int i = 0; i < 5; i++)
+        {
+            CountDownLatch latch = new CountDownLatch(1);
+            clientThreads.execute(() -> client.newRequest("localhost", connector.getLocalPort())
+                .send(result ->
+                {
+                    if (result.isSucceeded())
+                        latch.countDown();
+                }));
+            assertTrue(latch.await(1, TimeUnit.SECONDS));
+
+            // Check that the pool never grows above 1.
+            assertThat(poolRef.get().size(), is(1));
+
+            Thread.sleep(maxDuration * 2);
+        }
+
+        // Check that the pool acquired 5 and removed 4 connections;
+        // it must be exactly 4 removed b/c while the pool is not
+        // constrained, it can multiplex requests on a single connection
+        // so that should prevent opening more connections than needed.
+        assertThat(poolAcquireCounter.get(), is(5));
+        assertThat(poolRemoveCounter.get(), is(4));
```

Comment:
This test should assert creations, not acquisitions. But besides that, it asserts that multiplexed connections are not reused since they expire between each use. This is why the figures are the same as for the non-multiplexed tests.